dev/run-tests.py

```diff
@@ -477,7 +477,12 @@ def determine_java_version(java_exe):
 
     raw_output = subprocess.check_output([java_exe, "-version"],
                                          stderr=subprocess.STDOUT)
-    raw_version_str = raw_output.split('\n')[0]  # eg 'java version "1.8.0_25"'
+
+    raw_output_lines = raw_output.split('\n')
+
+    # find raw version string, eg 'java version "1.8.0_25"'
+    raw_version_str = next(str for str in raw_output_lines if " version " in str)
```

Comment:
Nit: `str` is a built-in function, so using it as a variable name will lead to spurious warnings in some Python linters.  I'd just use `x` or `s` instead.